### ISSUE TITLE
"You must either set a cache_path or a username" when using get_access_token

### ISSUE THREAD

**Issue by CharleyPearce** [plamere/spotipy#441]
I am using the oauth for the login system to my app, however I'm not providing a cache path because I am using a different form of caching the token info on my server.

When a user is redirected back from Spotify, I use `sp_oauth.get_access_token(code)` to verify the code. When this happens, I get the following error:
```
SpotifyOauthError at /app/
You must either set a cache_path or a username.
```
This is because `get_access_token` calls `get_cached_token`, which then throws an error when a cache path is not found.

Is it possible that I can call `get_access_token` to verify a code from Spotify without needing a specified cache path?

**Comment by stephanebruckert** [comment]
How are you creating the `sp_oauth` object at first? To provide the username, you should be able to:

 - either create `sp_oauth` by doing something like `sp_oauth = spotipy.SpotifyOAuth(username=your_username, etc)`
 - or by setting an environment variable `SPOTIPY_CLIENT_USERNAME`

**Comment by CharleyPearce** [comment]
I don't specify the username as when I define my sp_oauth object_object, I don't know the user's username at that point, so my definition will look something like this:
```
sp_oauth = oauth2.SpotifyOAuth( client_id, client_secret,redirect_uri,scope=scope)
```

**Comment by stephanebruckert** [comment]
Ha, I think I see. Currently the code is coupled to the fact that we want to cache the token as a file. The good news is we can change that.

First, can I suggest something as a temporary workaround? Can you try to provide a random cache_path or username:
 
     import random
     import string
     random = ''.join([random.choice(string.ascii_letters + string.digits) for n in xrange(32)])

     oauth2.SpotifyOAuth(client_id, client_secret, redirect_uri, scope=scope,
                         cache_path="/tmp/.cache-"+random)]

This will create a file that you will never use. It should allow you to retrieve the token using the code. Then I suppose you are storing this token somewhere else on your server, and the next time you want to use it, you could just do `sp = spotipy.Spotify(auth=token)`.

Let me know how this goes. If this works for you, we might be able to remove this warning and make a version that allows working without cache, or better, with custom caches.

**Comment by stephanebruckert** [comment]
Released in [2.9.0](https://github.com/plamere/spotipy/releases/tag/2.9.0), thanks for the quick & simple PR!
